strings/split.h

```diff
@@ -69,15 +69,30 @@ struct MatchImpl<std::string> {
 };
 
 template <size_t N>
-struct MatchImpl<const char[N]> {
+struct MatchImpl<const char[N]> {  // No `const char` because the calling semantics uses `rmconstref`.
```

Comment:
Hm, this comment looks definitely not obvious to me :)
1) [No `const char`] next to `const char`
2) [calling semantics uses `rmconstref`], but I can't see any `rmconstref` below